ext/bcmath/libbcmath/src/bcmath.h

```diff
@@ -79,6 +79,53 @@ typedef struct bc_struct {
 #define LONG_MAX 0x7ffffff
 #endif
 
+#if defined(_MSC_VER)
+#  include <stdlib.h>
+#  define BSWAP32(u) _byteswap_ulong(u)
+#  define BSWAP64(u) _byteswap_uint64(u)
+#else
+#  ifdef __has_builtin
+#    if __has_builtin(__builtin_bswap32)
+#      define BSWAP32(u) __builtin_bswap32(u)
+#    endif // __has_builtin(__builtin_bswap32)
+#    if __has_builtin(__builtin_bswap64)
+#      define BSWAP64(u) __builtin_bswap64(u)
+#    endif // __has_builtin(__builtin_bswap64)
+#  elif defined(__GNUC__) && ( \
+                    __GNUC__ > 4 || ( \
+                      __GNUC__ == 4 && ( \
+                        __GNUC_MINOR__ >= 3 \
+                      ) \
+                    ) \
+                  )
+#    define BSWAP32(u) __builtin_bswap32(u)
+#    define BSWAP64(u) __builtin_bswap64(u)
+#  endif // __has_builtin
+#endif // defined(_MSC_VER)
+
+#ifndef BSWAP32
+inline uint32_t BSWAP32(uint32_t u)
+{
+  return (((u & 0xff000000) >> 24)
+          | ((u & 0x00ff0000) >>  8)
+          | ((u & 0x0000ff00) <<  8)
+          | ((u & 0x000000ff) << 24));
+}
+#endif
```

Comment:
Why are we defining the 32bit version? This seems unused? Except if I missed something.

Reply:
Operation in a 32-bit environment had not yet been implemented. It is now implemented.